### PR TITLE
Manually flip deepep_mode for cuda_graph

### PR DESCRIPTION
**Pull request by zhuzilin**
## Motivation

When serving model with `--deepep-mode auto` and with cuda graph on, the current implementation won't flip the deepep mode from normal to low_latency and will trigger cuda illegal memory access in deepep or the following deepgemm kernel.

## Modifications

This PR is trying to add a manual flip by storing the deepep mode during capture and set it during replay.

Thank you for your time on reviewing this PR :)

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Comment by fzyzcjy** [comment]
ci passed

File: python/sglang/srt/model_executor/cuda_graph_runner.py [hunk 1]
```diff
@@ -655,6 +659,16 @@ def run_once():
...
             return logits_output_or_pp_proxy_tensors
 
+        # Resolve and pin DeepEP mode to avoid AUTO flipping between capture and replay
+        if get_moe_a2a_backend().is_deepep():
+            # Base runner handles decode/verify; not extend-in-batch
+            resolved = get_deepep_mode().resolve(is_extend_in_batch=False)
+            if resolved.is_low_latency():
+                DeepEPBuffer.set_dispatch_mode_as_low_latency()
+            else:
+                DeepEPBuffer.set_dispatch_mode_as_normal()
```
Comment (fzyzcjy) [inline python/sglang/srt/model_executor/cuda_graph_runner.py]: nit: extract to 

DeepEPBuffer.set_dispatch_mode(the mode variable)

File: python/sglang/srt/model_executor/cuda_graph_runner.py [hunk 2]
```diff
@@ -655,6 +659,16 @@ def run_once():
             )
             return logits_output_or_pp_proxy_tensors
 
+        # Resolve and pin DeepEP mode to avoid AUTO flipping between capture and replay
+        if get_moe_a2a_backend().is_deepep():
```
Comment (fzyzcjy) [inline python/sglang/srt/model_executor/cuda_graph_runner.py]: the logic of

```
def init:
  self._captured_deepep_mode = None

def capture:
  buffer.set dispatch mode
  self.mode = mode

def replay:
  buffer.set dispatch mode
```

seems to occur 3 times

thus I personally think we can have:

```
class DeepEPCudaGraphRunnerAdapter:
  def init
  def capture
  def replay
```

and thus in cudagraphrunner, eagle runner, eagle extend runner, we only need to add a field of `self.blahblah=DeepEPCudaGraphRunnerAdapter()` and call it

**Review by fzyzcjy — COMMENTED** [review]
LGTM, only nits

File: python/sglang/srt/layers/moe/token_dispatcher/deepep.py [hunk 3]
```diff
@@ -235,6 +235,13 @@ def set_dispatch_mode_as_low_latency(cls):
             cls.clean_buffer()
         cls._dispatch_mode = DeepEPDispatchMode.LOW_LATENCY
 
+    @classmethod
+    def set_dispatch_mode(cls, mode: DeepEPMode):
+        if mode.is_low_latency():
+            cls.set_dispatch_mode_as_low_latency()
+        else:
+            cls.set_dispatch_mode_as_normal()
```
Comment (fzyzcjy) [inline python/sglang/srt/layers/moe/token_dispatcher/deepep.py:243]:
```suggestion
        if mode.is_low_latency():
            cls.set_dispatch_mode_as_low_latency()
        elif mode.is_normal():
            cls.set_dispatch_mode_as_normal()
        else:
            raise Exception("unsupported mode")
```

**Review by fzyzcjy — APPROVED** [review]
LGTM
